wall_b.rb

```diff
@@ -146,3 +146,39 @@ class Wall
     # fix any errors.
   end
 end
+
+get '/walls/:wall_id' do
+  id = params[:wall_id]
+  @wall = Wall.get(id)
+  erb :show_wall
+end
+
+get '/walls/:wall_id/edit' do
+  id = params[:wall_id]
+  @wall = Wall.get(id)
+  erb :edit_wall
+end
+
+post '/walls/:wall_id/edit' do
+  id = params[:wall_id]
+  wall = Wall.get(id)
+  if params[:author] == wall.created_by
```

Comment:
Since you are using this logic of checking the author for your authentication, you could encapsulate the behavior more explicitly as an instance method for your `Wall` class:

```ruby
class Wall
  # ...
  def is_author?(author)
    author == self.author
  end
end
```

...and then this line could read like this:

```ruby
if wall.is_author?(params[:author])
  # ...
```